instrumentation/reactor/reactor-netty/reactor-netty-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/reactornetty/v1_0/FailedRequestWithUrlMaker.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.reactornetty.v1_0;
+
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
+import reactor.netty.http.client.HttpClientConfig;
+import reactor.netty.http.client.HttpClientRequest;
+
+final class FailedRequestWithUrlMaker {
+
+  static HttpClientRequest create(HttpClientConfig config, HttpClientRequest failedRequest) {
+    return (HttpClientRequest)
+        Proxy.newProxyInstance(
+            FailedRequestWithUrlMaker.class.getClassLoader(),
+            new Class<?>[] {HttpClientRequest.class},
+            new HttpRequestInvocationHandler(config, failedRequest));
+  }
+
+  private static final class HttpRequestInvocationHandler implements InvocationHandler {
+
+    private final HttpClientConfig config;
+    private final HttpClientRequest failedRequest;
+
+    private HttpRequestInvocationHandler(HttpClientConfig config, HttpClientRequest failedRequest) {
+      this.config = config;
+      this.failedRequest = failedRequest;
+    }
+
+    @Override
+    public Object invoke(Object proxy, Method method, Object[] args) throws Throwable {
+      if ("resourceUrl".equals(method.getName())) {
+        return computeUrlFromConfig();
+      }
+      try {
+        return method.invoke(failedRequest, args);
+      } catch (InvocationTargetException exception) {
+        throw exception.getCause();
+      }
+    }
+
+    private String computeUrlFromConfig() {
+      String uri = config.uri();
+      if (isAbsolute(uri)) {
+        return uri;
+      }
+
+      // use the baseUrl if it was configured
+      String baseUrl = config.baseUrl();
+      if (baseUrl != null) {
+        if (baseUrl.endsWith("/") && uri.startsWith("/")) {
+          baseUrl = baseUrl.substring(0, baseUrl.length() - 1);
```

Comment:
nit: the string containing a single forward slash causes an AIOOBE. Not sure if that can ever happen but it might be nice to guard against by also checking length > 1.

Reply:
Not sure if this can happen, but it's a nice catch either way 👍 
Thanks!